docs/developer/best_practices/commit.rst

```diff
@@ -0,0 +1,356 @@
+not need to check the entire commit message or the associated diff).
+It also helps the review process: if you have a big pull request affecting
+front-end and back-end, front-end people can only review commits starting with
+``ui`` (not need to read over the whole diff, or to open each commit one by one
+in Github to see which ones are interesting).
+
+Body
+::::
+
+The body should answer the following questions:
+
+- Why did you make this change? (is this for a new feature, a bugfix — then,
+  why was it buggy? —, some cleanup, some optimization, …).
+  It is really important to describe the intent/motivation behind the changes.
+
+- What change did you make? Document what the original problem was and how it
+  is being fixed (can be omitted for short obvious patches).
+
+- Why did you make the change in that way and not in another (mention alternate
+  solutions considered but discarded, if any)?
+
+When writing your message you must consider that your reader does not know
+anything about the code you have patched.
+
+You should also describe any limitations of the current code. This will avoid
+reviewer pointing them out, and also inform future people looking at the code
+which tradeoffs were made at the time.
+
+Lines should be wrapped at 72 characters.
+
+Footer
+::::::
+
+Use references such as ``Refs``, ``See``, ``Fixes`` or ``Closes`` followed by
+an issue number to automate issue management.
+
+Example:
+
+.. code:: text
+
+    topic: description
+
+    [… commit message body …]
+
+    Refs: #XXXXX, #YYYYY
+    Closes: #ZZZZZ
+
+Footer can also contains signature (``git commit -s``) or cherry-pick source
+(``git cherry-pick -x``).
+
+Examples
+^^^^^^^^
+
+Bad commit message
+::::::::::::::::::
+
+- ``Quick fix for service port issue``: what was the issue? It is a quick fix,
+  why not a proper fix? What are the limitations?
+
+- ``fix glitchs``: as expressive and useful as ~fix stuff~…
+
+- ``Bump Create React App to v3 and add optional-chaining``: Why? What are the
+  benefits?
+
+- ``Add skopeo & m2crypto to packages list``: Why do we need them?
+
+- ``Split certificates bootstrap between CA and clients``: Why do we need this
+  split? What is the issue we are trying to solve here?
+
+Note that none of these commits contain a reference to an issue (which
+could have been used as an (invalid) excuse for the lack of information): you
+really have no more context/explanation than what is shown here…
+
+Good commit message
+:::::::::::::::::::
+
+Commit `b531290c04c4 <https://github.com/scality/metalk8s/commit/b531290c04c45a01cd103a85431e2428b98d340e>`_
+''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+.. code:: text
+
+    Add gzip to nginx conf
+
+    This will decrease the size of the file the client need to download
+    In the current version we have ~7x improvement.
+    From 3.17Mb to 0.470Mb send to the client
```

Comment:
And this is exactly why we can't generate changelogs from the commits ;-)

Reply:
Indeed ^^